diff --git a/crypto/mbedtls/Make.defs b/crypto/mbedtls/Make.defs
index 3cb722d52..72dc8e4f5 100644
--- a/crypto/mbedtls/Make.defs
+++ b/crypto/mbedtls/Make.defs
@@ -23,10 +23,10 @@ CONFIGURED_APPS += $(APPDIR)/crypto/mbedtls
 
 # Allows `<mbedtls/<>.h>` import.
 
-CFLAGS += ${INCDIR_PREFIX}$(APPDIR)/crypto/mbedtls/include
-CFLAGS += ${INCDIR_PREFIX}$(APPDIR)/crypto/mbedtls/mbedtls/include
+CFLAGS += -isystem $(APPDIR)/crypto/mbedtls/include
+CFLAGS += -isystem $(APPDIR)/crypto/mbedtls/mbedtls/include
 
-CXXFLAGS += ${INCDIR_PREFIX}$(APPDIR)/crypto/mbedtls/include
-CXXFLAGS += ${INCDIR_PREFIX}$(APPDIR)/crypto/mbedtls/mbedtls/include
+CXXFLAGS += -isystem $(APPDIR)/crypto/mbedtls/include
+CXXFLAGS += -isystem $(APPDIR)/crypto/mbedtls/mbedtls/include
 
 endif
diff --git a/wireless/wapi/src/driver_wext.c b/wireless/wapi/src/driver_wext.c
index dc18610fe..d1fc2eecf 100644
--- a/wireless/wapi/src/driver_wext.c
+++ b/wireless/wapi/src/driver_wext.c
@@ -170,6 +170,7 @@ int wpa_driver_wext_set_key_ext(int sockfd, FAR const char *ifname,
       return -ENOMEM;
     }
 
+  memset(ext, 0, sizeof(*ext) + key_len);
   memset(&iwr, 0, sizeof(iwr));
   strlcpy(iwr.ifr_name, ifname, IFNAMSIZ);
 
@@ -202,6 +203,7 @@ int wpa_driver_wext_set_key_ext(int sockfd, FAR const char *ifname,
 
       default:
         nerr("ERROR: Unknown algorithm %d", alg);
+        memset(ext, 0, sizeof(*ext) + key_len);
         free(ext);
         return -EINVAL;
     }
@@ -212,6 +214,7 @@ int wpa_driver_wext_set_key_ext(int sockfd, FAR const char *ifname,
       nerr("ERROR: ioctl[SIOCSIWENCODEEXT]: %d", errno);
     }
 
+  memset(ext, 0, sizeof(*ext) + key_len);
   free(ext);
   return ret;
 }
